src/core/csv/reader_fread.cc

```diff
@@ -279,6 +279,7 @@ void FreadReader::detect_sep_and_qr() {
   // `numLines` has the number of lines in each group.
   int numFields[JUMPLINES+1];
   int numLines[JUMPLINES+1];
+  bool check_all_quoteRules = false;
```

Comment:
Maybe it's better to have variable `max_quoteRule`, which starts at 2 but then gets changed to 4 (or vice versa)?
Then  the for loop below could be  `for (quoteRule=0; quoteRule<max_quoteRule; quoteRule++)`.

Reply:
That's definitely more descriptive, done.